projects/components/src/overlay/modal/modal-ref.ts

```diff
@@ -0,0 +1,70 @@
+import { OverlayRef } from '@angular/cdk/overlay';
+import { Portal } from '@angular/cdk/portal';
+import { NavigationService } from '@hypertrace/common';
+import { Observable, Observer, ReplaySubject } from 'rxjs';
+import { filter, takeUntil } from 'rxjs/operators';
+
```

Comment:
This class doesn't need to mirror popoverRef - it can greatly simplify. To start, I'd probably strip it all the way down to something like:
```typescript
export class ModalRef<TResult> {
  public readonly closed$: Observable<TResult>;

  private readonly closedObserver: Observer<TResult>;

  public constructor(private readonly popoverRef: PopoverRef) {
    const closedSubject = new ReplaySubject<TResult>(1);
    this.closedObserver = closedSubject;
    this.closed$ = closedSubject.asObservable();
    this.popoverRef.closed$.subscribe({
      complete: () => this.closedObserver.complete(),
      error: err => this.closedObserver.error(err)
    });
  }

  public close(result?: TResult): void {
    if (result !== undefined) {
      this.closedObserver.next(result);
    }
    this.popoverRef.close();
  }
```